node/src/components/consensus/protocols/highway/performance_meter.rs

```diff
@@ -0,0 +1,147 @@
+use std::cmp::{max, min};
+
+use datasize::DataSize;
+use tracing::error;
+
+use casper_types::{TimeDiff, Timestamp};
+
+use crate::components::consensus::{
+    highway_core::{
+        finality_detector::{
+            assigned_weight_and_latest_unit, find_max_quora, round_participation,
+            RoundParticipation,
+        },
+        state, State,
+    },
+    traits::Context,
+    utils::ValidatorIndex,
+};
+
+/// The number of most recent blocks for which we average the max quorum in which we participated.
+const BLOCKS_TO_CONSIDER: usize = 5;
+/// The average max quorum that triggers us to slow down: with this big or smaller average max
+/// quorum per `BLOCKS_TO_CONSIDER`, we increase our round length.
+const SLOW_DOWN_THRESHOLD: f64 = 0.8;
+/// The average max quorum that triggers us to speed up: with this big or larger average max quorum
+/// per `BLOCKS_TO_CONSIDER`, we decrease our round length.
+const ACCELERATION_THRESHOLD: f64 = 0.9;
+/// We will try to accelerate (decrease our round length) every `ACCELERATION_PARAMETER` rounds if
+/// we have a big enough average max quorum.
+const ACCELERATION_PARAMETER: u64 = 40;
```

Comment:
im a little surprised that these are not chainspec settings.

Reply:
I probably forgot to remove the values from the config files, after we agreed that they should be hardcoded for now.
These are not chainspec'd, because it's not essential for the nodes to agree about them for the network to work. Every node could use its own method for determining when to switch the round exponent, and the network should still operate correctly.